plugin/src/main/kotlin/tanvd/grazi/spellcheck/GraziSpellchecker.kt

```diff
@@ -77,10 +79,20 @@ object GraziSpellchecker : GraziStateLifecycle {
      * Note, that casing and plural typos are ignored.
      */
     private fun check(word: String, project: Project, language: Language) = buildSet<Typo> {
-        val typo = tryRun { checker.check(word) }?.firstOrNull()?.let { Typo(it, checkerLang, 0) }
-        if (typo != null && IdeaSpellchecker.hasProblem(word, project, language)
-                && !isCasingProblem(word, typo) && !isPluralProblem(word, typo)) {
-            add(typo)
+        val typos = checkers.mapNotNull { (tool, checker) ->
+            tryRun { checker.match(tool.getAnalyzedSentence(word)) }?.firstOrNull()?.let { Typo(it, Lang[tool.language]!!, 0) }
+        }.filter { typo ->
+            !isCasingProblem(word, typo) && !isPluralProblem(word, typo) &&
+                    (IdeaSpellchecker.hasProblem(word, project, language) || !Text.isLatin(word))
+        }.toSet()
+
+        if (typos.size == checkers.size) {
```

Comment:
Hm, I would better check that set of languages of typos equals to checkers size. What if there will be more than one typo for one word? (just in case)

Reply:
Im not sure its possible, there are only one rule used for one word, but ok